File: src/main/java/org/hwabeag/playertime/expansions/InvClickEvent.java
package org.hwabeag.playertime.expansions;

import org.bukkit.ChatColor;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.hwabeag.playertime.config.ConfigManager;

import java.util.Objects;

public class InvClickEvent implements Listener {

    FileConfiguration PlayerTimeConfig = ConfigManager.getConfig("playertime");
    String Prefix = ChatColor.translateAlternateColorCodes('&', Objects.requireNonNull(PlayerTimeConfig.getString("playertime.prefix")));

    @EventHandler
    public void onClick(InventoryClickEvent e){
        if(e.getClickedInventory() == null)
            return;
        if(ChatColor.stripColor(e.getView().getTitle()).equalsIgnoreCase("TopPlayerTimeGUI")){
            e.setCancelled(true);
            Player player = (Player) e.getWhoClicked();
            player.sendMessage(Prefix + " 인벤토리 클릭이 취소되었습니다.");
        }
    }

}

File: src/main/kotlin/org/hwabeag/playertime/events/InvClickEvent.kt
package org.hwabeag.playertime.events

import org.bukkit.ChatColor
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryClickEvent
import org.hwabeag.playertime.PlayerTime

class InvClickEvent(private val plugin: PlayerTime) : Listener {

    @EventHandler
    fun onClick(event: InventoryClickEvent) {
        val clicked = event.clickedInventory ?: return
        if (ChatColor.stripColor(event.view.title) != "플레이타임 순위") {
            return
        }

        event.isCancelled = true
        val player = event.whoClicked as? Player ?: return
        if (clicked == player.inventory) {
            return
        }

        player.sendMessage("${plugin.configManager.prefix} 순위 창에서는 아이템을 이동할 수 없습니다.")
    }
}
